src/pages/build-apps/guides/data-storage.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Data storage
+description: Save and retrieve data for users with Gaia
+experience: beginners
+tags:
+  - tutorial
+images:
+  large: /images/pages/write-smart-contracts.svg
+  sm: /images/pages/write-smart-contracts-sm.svg
+---
+
+## Introduction
+
+This guide explains how to save and retrieve data for users with [Gaia](/build-apps/references/gaia) by implementing the [`connect`](https://github.com/blockstack/ux/tree/master/packages/connect#stacksconnect) and [`storage`](https://github.com/blockstack/ux/tree/master/packages/storage#stacksstorage) packages of Stacks.js.
+
+Data storage provides a way for users to save both public and private data off-chain while retaining complete control over it.
+
+Storing data off of the blockchain ensures that apps can provide users with high performance and high availability for data reads and writes without the involvement of centralized parties that could comprise their privacy or accessibility.
+
+See [the Todos app tutorial](/build-apps/tutorials/todos) for a concrete example of this functionality in practice.
+
+## Authentication
+
+TODO: Add indicator that authentication guide should be followed first
+
+## How data is stored
+
+Gaia storage is a key-value store.
```

Comment:
Something like "The key is a filename, and the value can be any type of data - images, text, video, etc.